test/java/seedu/addressbook/logic/CommandAssertions.java

```diff
@@ -29,6 +28,15 @@
     private static ExamBook examBook;
     private static StatisticsBook statisticsBook;
 
+    /**
+     * Enum used to indicate which storage file is the command targeting
+     * */
+    public enum TargetType {
+        AB,
+        EB,
+        SB
+    }
+
```

Comment:
A bit bad naming of enum though.
Shouldn't the enum be indicating what type of object the command is targeting, rather than what storage file?
So shouldn't it be Person, Exams, Stats etc?

Reply:
ok

Reply:
Hmm wait, if it is used only for single index command, wouldn't it be cleaner just to import the enum from there?

Reply:
No, its not just for indexes. Its just for input string and you expect something.

Reply:
Er. TargetType is only used for checking index of the thing targeted. It's only used in assertInvalidIndexBehaviour.
So I think it will be cleaner to import from IndexFormatCommand

Reply:
Also you shouldn't modify the base assertions, for example 
```
   @Test
    public void executeChangePasswordWrongPassword() throws Exception {
        final String initialPassword = addressBook.getMasterPassword();
        String expectedMessage = EditPasswordCommand.MESSAGE_WRONG_PASSWORD;
        assertCommandBehavior("editpw wrong_password new_password", expectedMessage,
                CommandAssertions.TargetType.PERSONS);
        assertCommandBehavior("editpw default_password1 new_password", expectedMessage,
                CommandAssertions.TargetType.PERSONS);
        assertCommandBehavior("editpw Default_password new_password", expectedMessage,
                CommandAssertions.TargetType.PERSONS);
        assertEquals(addressBook.getMasterPassword(), initialPassword);
    }

```
        assertCommandBehavior("editpw Default_password new_password", expectedMessage,
                CommandAssertions.TargetType.PERSONS);
doesn't make any sense as it is not targeting a person (or any of the enum for that matter).
So you need a assertCommandBehavior that doesn't require a TargetType also

Reply:
It is NOT for indexes. It is for checking which file it should be checking. The original one simply assumes an empty addressbook and checks for it accordingly in the methods called. However if I were to call a exam related command and expect an incorrect message, it should not check assert if the addressbook is indeed empty.

Reply:
Hence this is what I meant by it is targeting which file it is supposed to check instead of indexes such as what is documented  in indexformatcommand.

Reply:
Hence this is what I meant by it is targeting which file it is supposed to check instead of indexes such as what is documented  in indexformatcommand.

Reply:
by the way, what do you think of using assert(a.equals(b)) vs assertEquals(a,b)

Reply:
Ah. using the first gives merely an assertion error and has less info than the second.

Reply:
So, I have updated this method as well in the latest commit.

Reply:
Hmm I think we need to change this old code
```
public static void assertCommandBehavior(String inputCommand,
                                             String expectedStatusMessage,
                                             String expectedOutputMessage,
                                             AddressBook expectedAddressBook,
                                             boolean isRelevantPersonsExpected,
                                             List<? extends ReadOnlyPerson> lastShownList,
                                             boolean writesToFile) throws Exception {
        // If we need to test if the command writes to file correctly
        // Injects the saveFile object to check
        if (writesToFile) {
            logic.setStorage(saveFile);
        }
        //Execute the command
        CommandResult r = logic.execute(inputCommand);

        //Confirm the result contains the right data
        assertEquals(expectedOutputMessage, r.getOutputConsoleMessage());
        assertEquals(expectedStatusMessage, r.getStatusConsoleMessage());
        assertEquals(r.getRelevantPersons().isPresent(), isRelevantPersonsExpected);
        if (isRelevantPersonsExpected) {
            assertEquals(lastShownList, r.getRelevantPersons().get());
        }

        //Confirm the state of data is as expected
        assertEquals(expectedAddressBook, addressBook);
        assertEquals(lastShownList, logic.getLastShownList());
        if (writesToFile) {
            assertEquals(addressBook, saveFile.load());
        }
    }
```

Because the stub isn't used properly, because the actual storage file is injected at some point. Not sure how to use the stub though.

Reply:
I never understood why was a storage file injected haha.

Reply:
Ok. A stub is supposed to give a predetermined result and is used for Unit Testing. However if we want to test if it actually saves, that would be system testing.

Reply:
Hence so far, from this, I think that whenever writesToFile is false, storage is a StorageStub i.e. empty.

Reply:
It seems that we're doing two types of test at the same time.

Reply:
If we are removing storagefile, can refer to #215 and I'll close this PR.